proxy.go

```diff
@@ -124,37 +123,30 @@ func linkUp() error {
 
 func rx(conn net.Conn, tap *water.Interface, errCh chan error, mtu int) {
 	elog.Println("Waiting for frames from enclave application.")
-	var frame ethernet.Frame
+	frame := make([]byte, mtu)
+	size := make([]byte, 2)
 	for {
-		frame.Resize(mtu)
 		n, err := tap.Read([]byte(frame))
 		if err != nil {
 			errCh <- fmt.Errorf("failed to read packet from TAP device: %w", err)
 			return
 		}
-		frame = frame[:n]
 
-		size := make([]byte, 2)
 		binary.LittleEndian.PutUint16(size, uint16(n))
-
-		if _, err := conn.Write(size); err != nil {
-			errCh <- fmt.Errorf("failed to write frame size to connection: %w", err)
-			return
-		}
-		if _, err := conn.Write(frame); err != nil {
-			errCh <- fmt.Errorf("failed to write frame to connection: %w", err)
+		if _, err := conn.Write(append(size, frame[:n]...)); err != nil {
```

Comment:
Does this copy `frame[:n]` into `size`? If so that's allocating again, I think.

Would it work to `frame := make([]byte, mtu+2)` outside the loop, then read into `frame[2:]` and splat the length into `frame[:2]` before sending the whole thing? That way you wouldn't need a separate slice for the size.

Reply:
Thanks for pointing this out. 969c14c fixes the issue (basically by doing what you suggested), along with some refactoring, and an increase of the send and receive buffer.

While doing another round of performance tests, I noticed a drop in reqs/sec when sending from ~500 concurrent threads. I therefore bumped `MaxIdleConns` and `MaxIdleConnsPerHost` in edfda01.

With all these fixes in place, we can now sustain ~5K reqs/sec for 512 sending threads, and probably more as the senders increase.